src/components/Inputs/Toggle.tsx

```diff
@@ -17,31 +17,44 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { motion, Transition } from 'framer-motion'
+import { useEffect, useState } from 'react'
 import styled, { css, useTheme } from 'styled-components'
 
 interface ToggleProps {
   toggled: boolean
   onToggle: (value: boolean) => void
   className?: string
   disabled?: boolean
+  ToggleIcons?: [LucideIconType, LucideIconType]
+  handleColors?: [string, string]
 }
 
 const toggleMarginInPx = 2
 
-const Toggle = ({ toggled, onToggle, className, disabled }: ToggleProps) => {
+const Toggle = ({ toggled, onToggle, className, disabled, ToggleIcons, handleColors }: ToggleProps) => {
   const theme = useTheme()
-  const toggleWidth = parseInt(getComputedStyle(document.documentElement).getPropertyValue('--toggleWidth'))
+  const [toggleWidth, setToggleWidth] = useState(0)
+  const [ToggleIconRight, ToggleIconLeft] = ToggleIcons ?? [undefined, undefined]
 
   const toggleBackgroundVariants = {
     off: { backgroundColor: theme.bg.tertiary },
-    on: { backgroundColor: theme.global.accent }
+    on: { backgroundColor: handleColors ? theme.bg.tertiary : theme.global.accent }
   }
 
-  const floatingIndicatorVariant = {
+  const handleContainerVariants = {
     off: { left: 0 },
     on: { left: toggleWidth / 2 - toggleMarginInPx * 2 }
   }
 
+  const handleVariants = {
+    off: { backgroundColor: handleColors?.[0] ?? 'var(--color-white)' },
+    on: { backgroundColor: handleColors?.[1] ?? 'var(--color-white)' }
+  }
+
+  useEffect(() => {
+    setToggleWidth(parseInt(getComputedStyle(document.documentElement).getPropertyValue('--toggleWidth')))
+  }, [])
```

Comment:
If my `useEffect` intuition is correct, this will only ever fire once since the dependency list is empty? Intended or should it be omitted?

Reply:
Yes! I want this to be ran on first render only 👍